bookkeeper-server/src/main/java/org/apache/bookkeeper/client/PendingAddOp.java

```diff
@@ -253,6 +251,7 @@ void submitCallback(final int rc) {
             addOpLogger.registerSuccessfulEvent(latencyNanos, TimeUnit.NANOSECONDS);
         }
         cb.addComplete(rc, lh, entryId, ctx);
+        ReferenceCountUtil.release(toSend);
```

Comment:
I don't think this change is necessary. We can release the before before trigger the callback, since we are not going to touch it anymore at this point.